batch/batch/front_end/front_end.py

```diff
@@ -408,47 +408,61 @@ def has_resource_available(record):
     return True
 
 
-def attempt_id_from_spec(record):
+def attempt_id_from_spec(record) -> Optional[str]:
     return record['attempt_id'] or record['last_cancelled_attempt_id']
 
 
-async def _get_job_log(app, batch_id, job_id):
-    record = await _get_job_record(app, batch_id, job_id)
+async def _get_job_container_log_from_worker(client_session, batch_id, job_id, container, ip_address) -> bytes:
+    try:
+        async with await request_retry_transient_errors(
+            client_session,
+            'GET',
+            f'http://{ip_address}:5000/api/v1alpha/batches/{batch_id}/jobs/{job_id}/log/{container}',
+        ) as resp:
+            return await resp.read()
```

Comment:
Just so I understand, the FileResponse here doesn't deserve special treatment as it's just a special case of a regular Response?

Reply:
Yes, I think it subclasses `Response`. The client can't tell, but it's an efficient way for the server to stream bytes directly from an open file handle instead of loading the whole thing into memory on the server and then sending those bytes.